core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -1252,6 +1254,28 @@ private[spark] class DAGScheduler(
     execCores.map(cores => properties.setProperty(EXECUTOR_CORES_LOCAL_PROPERTY, cores))
   }
 
+  /**
+   * If push based shuffle is enabled, set the shuffle services to be used for the given
+   * shuffle map stage. The list of shuffle services is determined based on the list of
+   * active executors tracked by block manager master at the start of the stage.
+   */
+  private def prepareShuffleServicesForShuffleMapStage(stage: ShuffleMapStage) {
+    // TODO: Handle stage reuse/retry cases separately as without finalize changes we cannot
```

Comment:
Maybe, we can add a shuffle service master role to manage the shuffle service worker. App driver requests shuffle service workers from shuffle service master. The master can customize the politics to assign the workers for apps. Also, the master manages the blacklist of shuffle service worker, adding or removing shuffle workers. In addition, the master can notify worker to delete data of completed apps by heart beat.

Reply:
@JulySouthWind I think you might have commented here by mistake. The relevant PR that is actively being worked on is https://github.com/apache/spark/pull/30164.
This was a linkedin internal PR that was created but now we will just close this.